src/domain/organizations/organizations-safe.repository.spec.ts

```diff
@@ -0,0 +1,679 @@
+          getStringEnumKeys(OrganizationStatus),
+        ),
+        name: faker.word.noun(),
+      });
+      const orgId = org.identifiers[0].id as Organization['id'];
+      await dbUserOrgRepo.insert({
+        user: { id: userId },
+        role: 'ADMIN',
+        status: 'ACTIVE',
+        organization: { id: orgId },
+      });
+
+      await orgSafesRepo.create({
+        organizationId: orgId,
+        payload: [
+          {
+            chainId,
+            address,
+          },
+        ],
+      });
+
+      await expect(
+        dbOrgSafeRepo.find({
+          where: { chainId, address },
+        }),
+      ).resolves.toEqual([
+        {
+          id: expect.any(Number),
+          chainId,
+          address,
+          createdAt: expect.any(Date),
+          updatedAt: expect.any(Date),
+        },
+      ]);
+    });
+
+    it('should create multiple OrganizationSafes', async () => {
+      const payload = faker.helpers.multiple(
+        () => ({
+          chainId: faker.string.numeric(),
+          address: getAddress(faker.finance.ethereumAddress()),
+        }),
+        { count: { min: 2, max: 5 } },
+      );
+      const user = await dbUserRepo.insert({
+        status: 'ACTIVE',
+      });
+      const userId = user.identifiers[0].id as User['id'];
+      await dbWalletRepo.insert({
+        user: { id: userId },
+        address: getAddress(faker.finance.ethereumAddress()),
+      });
+      const org = await dbOrgRepo.insert({
+        status: faker.helpers.arrayElement(
+          getStringEnumKeys(OrganizationStatus),
+        ),
+        name: faker.word.noun(),
+      });
+      const orgId = org.identifiers[0].id as Organization['id'];
+      await dbUserOrgRepo.insert({
+        user: { id: userId },
+        role: 'ADMIN',
+        status: 'ACTIVE',
+        organization: { id: orgId },
+      });
+
+      await orgSafesRepo.create({
+        organizationId: orgId,
+        payload,
+      });
+
+      await expect(dbOrgSafeRepo.find()).resolves.toEqual(
+        payload.map(({ chainId, address }) => ({
+          id: expect.any(Number),
+          chainId,
+          address,
+          createdAt: expect.any(Date),
+          updatedAt: expect.any(Date),
+        })),
+      );
+    });
+  });
+
+  describe('findByOrganizationId', () => {
+    it('should return found organization Safes', async () => {
+      const orgSafes = faker.helpers.multiple(
+        () => ({
+          chainId: faker.string.numeric(),
+          address: getAddress(faker.finance.ethereumAddress()),
+        }),
+        { count: { min: 2, max: 5 } },
+      );
+      const org = await dbOrgRepo.insert({
+        status: faker.helpers.arrayElement(
+          getStringEnumKeys(OrganizationStatus),
+        ),
+        name: faker.word.noun(),
+      });
+      const orgId = org.identifiers[0].id as Organization['id'];
+      await orgSafesRepo.create({
+        organizationId: orgId,
+        payload: orgSafes,
+      });
```

Comment:
Should we rely on another method here, or insert the entry via the database repo?

Reply:
I've changed `orgSafesRepo.create` by `dbOrgSafeRepo.insert` here :)